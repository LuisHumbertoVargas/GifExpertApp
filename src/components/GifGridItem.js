import React from 'react';

const GifGridItem = ({ id, url, title }) => {
    return (
        <div className='card animate__animated animate__backInRight'>
            <img src={url} alt={title} />
            <p>{title}</p>
            {/* <p>ID: {id}</p> */}
        </div>
    );
};

export default GifGridItem;
