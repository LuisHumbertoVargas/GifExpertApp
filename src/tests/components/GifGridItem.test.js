import React from 'react';
import { shallow } from 'enzyme';
import GifGridItem from '../../components/GifGridItem';
import PropTypes from 'prop-types';

describe('Prueba unitaria en el componente <GifGridItem />', () => {
    const title = 'TÃ­tulo del Gif';
    const url = 'https://localhost/algo.jpg';
    const wrapper = shallow(<GifGridItem title={title} url={url} />);

    test('Debe mostrar el componente correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de tener una etiqueta <img></img> con sus atributos src y alt', () => {
        const img = wrapper.find('img');
        // console.log(img.html());
        // console.log(img.props().src);
        // console.log(img.prop('src'));
        expect(img.prop('alt')).toBe(title);
        expect(img.prop('src')).toBe(url);
    });

    test('Debe de tener una etiqueta <p>con el title</p>', () => {
        const p = wrapper.find('p').text().trim();
        expect(p).toBe(title);
    });

    test('Debe de tener la clase animate__backInRight', () => {
        const div = wrapper.find('div');
        const clase = div.props().className;
        expect(clase.includes('animate__backInRight')).toBe(true);
    });
});

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
};
