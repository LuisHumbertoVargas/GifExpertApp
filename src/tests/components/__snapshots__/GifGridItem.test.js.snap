// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba unitaria en el componente <GifGridItem /> Debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__backInRight"
>
  <img
    alt="Título del Gif"
    src="https://localhost/algo.jpg"
  />
  <p>
    Título del Gif
  </p>
</div>
`;
